docs/guide/env-and-mode.md

```diff
@@ -16,6 +16,31 @@ Some built-in constants are available in all cases:
 
 - **`import.meta.env.SSR`**: {boolean} whether the app is running in the [server](./ssr.md#conditional-logic).
 
+## Conditional Compilation
```

Comment:
I don't think we need a whole section for this. I think a simple example is enough.

````markdown
:::details Example
```js
if (import.meta.env.DEV) {
  // code inside here will be treeshaken
  console.log('Dev mode') 
}
```
:::
````